modules/mwi/mwi.c

```diff
@@ -154,7 +156,8 @@ static void ua_event_handler(struct ua *ua,
 
 	if (ev == UA_EVENT_REGISTER_OK) {
 
-		if (!mwi_find(ua))
+		if (!mwi_find(ua) &&
+		    (strcmp(account_mwi(ua_account(ua)), "yes") == 0))
```

Comment:
please be aware that `strcmp` does not handle NULL input, in which case it will crash.

it is better to use `str_casecmp` or let the `account_mwi` function return `bool`.